src/prefect/flows.py

```diff
@@ -158,6 +159,28 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hook passed is list and contains callables
+        hook_categories = [on_completion, on_failure, on_cancellation, on_crashed]
+        hook_names = ["on_completion", "on_failure", "on_cancellation", "on_crashed"]
+        for hooks, hook_name in zip(hook_categories, hook_names):
+            if hooks is not None:
+                if not hooks:
+                    raise ValueError(f"Empty list passed for '{hook_name}'")
+                try:
+                    hooks = list(hooks)
+                except TypeError:
```

Comment:
@rkscodes, this is looking great! The only thing I can think of is to try to avoid a dead end when raising this `TypeError.` It would be more helpful to show the user what is expected:

```python
            raise TypeError(
                f"Expected iterable for '{hook_name}'; got"
                f" {type(hooks).__name__} instead. Please provide a list of hooks to 
                f"{hook_name}. For example:"
                f"\n@flow(on_completion=[hook1, hook2])\ndef flow1():\npass"
            )
```

Reply:
Yeah that would be more helpful for the user. I will adding the same msg looks good to me.